app/db/models.py

```diff
@@ -59,15 +79,19 @@ class DBPrivateAIKey(Base):
     litellm_api_url = Column(String)
     owner_id = Column(Integer, ForeignKey("users.id"))
     region_id = Column(Integer, ForeignKey("regions.id"))
-    created_at = Column(DateTime, default=datetime.utcnow)
+    created_at = Column(DateTime(timezone=True), default=func.now())
+    updated_at = Column(DateTime(timezone=True), onupdate=func.now())
+    team_id = Column(Integer, ForeignKey("teams.id"))
 
     owner = relationship("DBUser", back_populates="private_ai_keys")
     region = relationship("DBRegion", back_populates="private_ai_keys")
+    team = relationship("DBTeam", back_populates="private_ai_keys")
```

Comment:
in other projects I've found it useful to have a `BaseTeamModel` class that has all the fields needed to belong to a team. this makes it a bit more explicit when you want to handle things like permission checks in a generic/central way. May not be worth it for this project, though, especially if each one needs a unique `back_populates`...